app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -0,0 +1,99 @@
+package com.example.bottomnav;
+
+import org.eclipse.cdt.core.dom.ast.IASTDeclaration;
+import org.eclipse.cdt.core.dom.ast.IASTNode;
+import org.eclipse.cdt.core.dom.ast.IASTTranslationUnit;
+import org.eclipse.cdt.core.dom.ast.gnu.cpp.GPPLanguage;
+import org.eclipse.cdt.core.index.IIndex;
+import org.eclipse.cdt.core.parser.*;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Parse {
+    public HashMap<String, Contents> constRepo = new HashMap<>();
+    public HashMap<String, ArrayList> structRepo = new HashMap<>();
+    private ArrayList subStrucCon = null;
+
+    public Parse(String code) throws Exception {
+        IASTTranslationUnit translationUnit = getIASTTranslationUnit(code.toCharArray());
+        IASTNode node = translationUnit;
+        saveNodes(node);
```

Comment:
Will this work instead?
```suggestion
        saveNodes(translationUnit);
```